packages/astro/src/core/dev/index.ts

```diff
@@ -52,6 +52,9 @@ export default async function dev(
 			optimizeDeps: {
 				include: rendererClientEntries,
 			},
+			define: {
+				'import.meta.env.BASE_URL': settings.config.base ? `'${settings.config.base}'` : 'undefined',
```

Comment:
Would it be the same as the following?

```
settings.config.base || 'undefined'
```

Reply:
Refer to Vite Define docs: https://vitejs.dev/config/shared-options.html#define

and https://github.com/withastro/astro/blob/main/packages/astro/src/core/create-vite.ts#L96

We do need to use strings.

Reply:
And If base is not set, it would show '/' in that case here

Reply:
If base is not set, it would show `/` in that case here.

What's more, `settings.config.base` and  `'${settings.config.base}'`</span> are different, notice the **single quotes**

Reply:
@yuhang-dong yup, now I see it. Thanks for the accurate pointers!